p2p/pex/pex_reactor.go

```diff
@@ -170,12 +170,18 @@ func (r *PEXReactor) AddPeer(p Peer) {
 		}
 	} else {
 		// inbound peer is its own source
-		addr := p.SocketAddr()
+		addr, err := p.NodeInfo().NetAddress()
```

Comment:
That's the line fixing the problem, right?

Reply:
Exactly. Funny enough the `connection refused` errors popped up in the pex tests, unfortunately in a context where we already expected the connection to fail.